src/modules/user/user.controller.ts

```diff
@@ -835,4 +841,72 @@ export class UserController {
   async verifyAccountEmail(@Body() body: VerifyEmailDto) {
     return this.userUseCases.verifyUserEmail(body.verificationToken);
   }
+
+  @Patch('/profile')
+  @HttpCode(200)
+  @ApiBearerAuth()
+  @ApiOperation({
+    summary: 'Update user profile',
+  })
+  @ApiOkResponse({
+    description: 'Updated user profile',
+  })
+  async updateProfile(
+    @UserDecorator() user: User,
+    @Body() updateProfileDto: UpdateProfileDto,
+  ) {
+    if (!updateProfileDto.name && updateProfileDto.lastname == undefined) {
```

Comment:
Making the `lastname` optional and the `name` not makes this check unnecesary. As long as the requester sends `name` or `name` and `lastname`, everything should be fine